### PR TITLE
Fix TypeError in container's wait method

### PR DESCRIPTION
**Pull request by mrbazzan**
For the doc change: list of string does not work with `environment` argument

Fixes No.3 in #184, `.wait()` returns an integer not a dictionary

### CODE REVIEW

**Comment by rhatdan** [comment]
LGTM
@mwhahaha @jwhonce @cdoern PTAL

File: podman/domain/containers_run.py [hunk 1]
```diff
@@ -77,7 +77,7 @@ def run(
         if log_type in ("json-file", "journald"):
             log_iter = container.logs(stdout=stdout, stderr=stderr, stream=True, follow=True)
 
-        exit_status = container.wait()["StatusCode"]
+        exit_status = container.wait()
         if exit_status != 0:
```
Comment (cdoern) [inline podman/domain/containers_run.py:81]: how does this check still work if you remove the ["StatusCode"] aspect? now you are checking `if DICT != 0`

Comment (rhatdan) [reply podman/domain/containers_run.py:81]: def wait(self, **kwargs) -> Dict[Literal["StatusCode", "Error"], Any]:
Looks like to original code is correct

Comment (sylencecc) [reply podman/domain/containers_run.py:81]: Wait wait wait, there is definitely an issue here.
By default, `wait()` in podman-py uses the libpod operation  [ContainerwaitLibPod](https://docs.podman.io/en/latest/_static/api.html#tag/containers/operation/ContainerWaitLibpod), which returns a JSON-encoded integer. For that case, the type annotation
```
def wait(self, **kwargs) -> Dict[Literal["StatusCode", "Error"], Any]:
```
isn't correct. That annotation matches the podman compat API, where [ContainerWait](https://docs.podman.io/en/latest/_static/api.html#tag/containers-(compat)/operation/ContainerWait) returns a JSON object with the expected Keys `Error` and `StatusCode`.
I just discovered podman-py and don't know the code base enough yet to fix it myself, but would recommend to look into that issue again. I got here having experienced the same issue as in #201

Comment (rhatdan) [reply podman/domain/containers_run.py:81]: @cdoern PTAL

**Comment by jonathanunderwood** [comment]
@rhatdan wondering why this was closed, as the issue is still present. As @sylencecc points out, the proposed fix is correct, but the type annotation for the return type of`wait()` is wrong and also needs fixing.

**Comment by rhatdan** [comment]
I don't remember but might have been lack of activity.

Care to open a new PR based on this?

**Pull request closed without merging**